src/components/node-list/styles/_filter.scss

```diff
@@ -14,72 +39,89 @@
     fill: none;
     stroke: var(--color-nodelist-filter-indicator-off);
   }
-}
 
-// Fade row text when unchecked
-.pipeline-nodelist__row--unchecked .pipeline-nodelist__row__text--kind-filter {
-  opacity: 0.55;
+  .pipeline-nodelist__row:hover &.pipeline-row__toggle-icon--all-unset,
+  [data-whatintent='keyboard'] .pipeline-nodelist__row__checkbox:focus + &,
+  &.pipeline-row__toggle-icon--parent {
+    fill: $color-primary;
+    stroke: $color-primary;
+  }
 }
 
-// Brighter row text when unchecked and hovered
-.pipeline-nodelist__row--unchecked:hover
-  .pipeline-nodelist__row__text--kind-filter {
-  opacity: 0.8;
-}
+//--- Toggle icon opacities ---//
```

Comment:
quick thought: while we are making the effort to properly set up the style rules, I wonder whether it's worth to separate the style classes for the toggle button out from `_filter.scss` to something like `_toggle.scss`?

Reply:
We could do, but some of those ones currently apply to all of the node-list rows, and these override them. There's only a couple of rules that apply there, I'm not sure that it's necessary to have a new file for just the `--kind-toggle` CSS.

_However_, I think there's a decent case to be made for establishing new files for the row-label and row-icon CSS, so that the overrides are all in one place. Could do, but it makes reviewing this PR more difficult as all the styles would be copied to a new file so it's more difficult to compare them, so if we were to do that I'd be inclined to do it in a separate PR.

Reply:
sounds good, the process of migrating those style rules can be done in a seperate PR ( it might be worth to even combine this with the future work in the refactor of the sidebar to allow more generic data as props)